Sources/TuistGenerator/Generator/WorkspaceGenerator.swift

```diff
@@ -84,17 +80,31 @@ final class WorkspaceGenerator: WorkspaceGenerating {
         let workspaceName = "\(graph.name).xcworkspace"
 
         Printer.shared.print(section: "Generating workspace \(workspaceName)")
-
+        
         /// Projects
 
         var generatedProjects = [AbsolutePath: GeneratedProject]()
-        try graph.projects.forEach { project in
-            let generatedProject = try projectGenerator.generate(project: project,
-                                                                 graph: graph,
-                                                                 sourceRootPath: project.path)
-            generatedProjects[project.path] = generatedProject
+        
+        let ignoredProjects: [AbsolutePath]
+        
+        if CLI.arguments.carthage.projects {
+            ignoredProjects = [ ]
+        } else {
+            ignoredProjects = graph.carthageDependencies.map(\.projectPath)
         }
-
+        
+        let workspace = Workspace(
+            name: workspaceName,
+            projects: workspace.projects.filter(ignoredProjects.doesNotContain),
+            additionalFiles: workspace.additionalFiles
+        )
+        
+        for project in graph.projects where ignoredProjects.doesNotContain(project.path) {
+            generatedProjects[project.path] = try projectGenerator.generate(project: project,
+                                                                            graph: graph,
+                                                                            sourceRootPath: project.path)
+        }
```

Comment:
This change is required so that we do not generate carthage projects - ideally I would like to do this in a nicer place away from the workspace.

Reply:
Moved to TargetNode.node where CarthageNode is defined.